package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -20,24 +21,65 @@ def get_catalog(self) -> DataCatalog:
     def set_catalog(self, value: DataCatalog):
         self._catalog = value
 
+    def get_layers(self) -> Optional[Dict[str, Set[str]]]:
+        """
+        This method retrieves the layers from the catalog. For kedro-datasets versions before 1.3.0,
+        the 'layers' attribute is defined directly in the catalog.
+
+        From kedro-datasets 1.3.0 onwards, the 'layers' attribute is defined inside the 'metadata'
+        under 'kedro-viz' plugin.
+
+        Catalog before kedro-datasets 1.3.0:
+            type: pandas.CSVDataSet
+            filepath: /filepath/to/dataset
+            layers: raw
+
+        Catalog from kedro-datasets 1.3.0 onwards:
+            type: pandas.CSVDataSet
+            filepath: /filepath/to/dataset
+            metadata:
+                kedro-viz:
+                    layers: raw
+
+        Currently, Kedro up to 18.x and kedro-viz up to 6.x support both formats.
+        However, support for the old format will be discontinued from Kedro 19.x and kedro-viz
+        7.x onwards.
+        It's recommended to follow the newest format for defining layers in the catalog.
+        """
+        if self._catalog.layers:
+            return self._catalog.layers
+
+        layers: Dict[str, Set[str]] = defaultdict(set)
+
+        for dataset_name in self._catalog._data_sets:
+            dataset = self._catalog._get_dataset(dataset_name)
+            metadata = getattr(dataset, "metadata", {}) or {}
```

Comment:
What is the `or` condition for? The third argument in `getattr` is already the default return if `metadata` isn't found.

Reply:
it's to handle a case where metadata = None. When I was working with preview function, this error got thrown multiple times so just handling it here. 
I guess this is because we define metadata as None by default here - https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/kedro_datasets/pandas/csv_dataset.py#L80